### DISCUSSION TITLE
displayname in Datagrid

### DISCUSSION THREAD

**Discussion by kyotr** [Megabit/Blazorise#5220] [Q&A] [answered]
Hello,

is it possible to retrieve display name from my model and use it in Datagrid  Column Caption ?
Example :
```
     [ Display(Name = "Account Name")]    
    public string Name { get; set; }
```

>          <DataGridColumn Field="@nameof(Account.Name)" Caption="@Account.Name" Editable />
 
or  in label : 
 `<InputText` class="form-control" DisplayName="Account.Name" @bind-Value="@Account.Name" />

Thanks

**Comment by David-Moreira** [comment]
I have something similar to what you want around here:
In this case it's using the **DisplayNameAttribute**.
```

        public static string GetDisplayName<T>(string propertyName)
        {
            var propertyInfo = typeof(T).GetProperty(propertyName);
            if (propertyInfo is null)
                throw new System.ArgumentException($"Could not find {propertyName} on type: {typeof(T).Name}");

            var displayName = propertyInfo.GetCustomAttribute<DisplayNameAttribute>();
            return displayName is null
                ? propertyName
                : displayName.DisplayName;
            //throw new System.ArgumentException($"No display name attribute found for: {propertyName}")
        }

```


You could also go further and make your own grid column based on ours that automatically gets the caption from the attribute based on the Field that's been set.

**Reply by kyotr** [reply]
Thank you for your answer 

i have tried this, (i guess i made a mistake here)

`                 <DataGridColumn Field="@nameof(ProductBrandVM.ProductBrandName)" Validator="@CheckName" Caption=**@GetDisplayName<ProductBrandVM>('ProductBrandName')** Editable />
`

But i m getting this message 

`cannot convert from 'method group' to 'string'`


 then i have used this

```
    public string GetDisplayName(string Name)

    {
        try
        {


            var name = "";

            MemberInfo property = typeof(ProductBrandVM).GetProperty(Name);

            var dd = property.GetCustomAttribute(typeof(DisplayAttribute)) as DisplayAttribute;

            if (dd != null)

            {

                name = dd.Name;

            }

            return name;
        }
        catch (Exception e)
        {
            NotificationService.Error("Error: " + Name, e.Message);
            return Name;
        }

        return Name;

    }
```

**Reply by David-Moreira** [reply]
You hardcoded the type here : `typeof(ProductBrandVM).GetProperty(Name);`. The idea is that you could use with any type.
Anyway, what you mean, is that it worked? Correct? Great.

**Reply by kyotr** [reply]
Yes, thats why i like your method. is it possible to know what i did wrong here ?

`                 <DataGridColumn Field="@nameof(ProductBrandVM.ProductBrandName)" Caption=@GetDisplayName<ProductBrandVM>("ProductBrandName") Editable />
`

**Reply by David-Moreira** [reply, marked as answer]
The code seems ok from looking at it. Try `Caption=@(GetDisplayName<ProductBrandVM>("ProductBrandName"))` just in case.

**Reply by David-Moreira** [reply]
The '<' might be confusing the compiler, so **@** might help if you are having an error thrown by the razor compiler.

**Reply by kyotr** [reply]
Many thanks, it is working now
